src/main/java/datawave/query/util/FrequencyFamilyCounter.java

```diff
@@ -0,0 +1,135 @@
+ * out of Accumulo.
+ */
+
+public class FrequencyFamilyCounter {
+    
+    private TreeMap<YearMonthDay,Frequency> dateToFrequencyValueMap = new TreeMap<>();
+    private static final int SIMPLEDATE_LENGTH = 8;
+    private DateFrequencyValue serializer = new DateFrequencyValue();
+    
+    private static final Logger log = LoggerFactory.getLogger(FrequencyFamilyCounter.class);
+    
+    public FrequencyFamilyCounter() {
+        
+    }
+    
+    public void initialize(Value value) {
+        deserializeCompressedValue(value);
+    }
+    
+    public void clear() {
+        dateToFrequencyValueMap.clear();
+    }
+    
+    public TreeMap<YearMonthDay,Frequency> getDateToFrequencyValueMap() {
+        return dateToFrequencyValueMap;
+    }
+    
+    /**
+     * Takes the value of a compressed f column record and creates the qualifierToFrequencyValueMap so that other records that are not yet aggregated can be
+     * added to the compressed record. After those records are aggregated they are discarded.
+     *
+     * @param oldValue
+     */
+    public void deserializeCompressedValue(Value oldValue) {
+        dateToFrequencyValueMap = serializer.deserialize(oldValue);
+    }
+    
+    /**
+     * Performs what the SummingCombiner used to perform for date and frequency values. The function accumulutes ingest frequencies for individual dates
+     */
+    
+    public void aggregateRecord(String key, String value) {
+        
+        insertIntoMap(key, value);
+    }
+    
+    /**
+     * Inserts a key and value into the qualifiedToFrequencyValueMap and converts the string value to a long
+     * 
+     * @param key
+     *            The value for "key" is the old style ColumnQualifier of this format: csv\x0020160426
+     * @param value
+     *            value is a an Integer in string format that might be hexadecimal, decimal or octal
+     */
+    public void insertIntoMap(String key, String value) {
+        int parsedInteger;
+        String cleanKey;
+        
+        // Assuming that as SimpleDate is at the end of the key passed in. yyyyMMdd
+        // The value for "key" is the old style ColumnQualifier of this format:
+        // csv\x0020160426 and the value is a an Integer in string format that
+        // might be hexadecimal, decimal or octal
+        if (key != null) {
+            if (key.length() > SIMPLEDATE_LENGTH) {
+                cleanKey = key.substring(key.length() - SIMPLEDATE_LENGTH);
+            } else {
+                cleanKey = key;
+            }
+        } else
+            return;
+        
+        if (log.isTraceEnabled())
+            log.trace("inserting key: " + cleanKey + " value: " + value);
+        
+        if (value.isEmpty())
+            return;
+        
+        try {
+            parsedInteger = Integer.parseUnsignedInt(value);
```

Comment:
I found this to fail working with aggregated values.  When is this class supposed to be used?

Reply:
It's a helper class that holds the aggregated DateFrequencyValue object and manages it.  Maybe the counts are getting too high in your data?  I used to process Longs but we decided to contract the values to Integer. Shouldn't be hard to increase the size of the count value.

Reply:
I pushed the code changes for the other comments.